tests/integration/test_examples_on_kubeflow.py

```diff
@@ -0,0 +1,348 @@
+    module_mocker.patch(
+        "zenml.integrations.kubeflow.orchestrators.local_deployment_utils.start_kfp_ui_daemon"
+    )
+
+    tmp_path = tmp_path_factory.mktemp("tmp")
+    os.chdir(tmp_path)
+    Repository.initialize(root=tmp_path)
+    repo = Repository(root=tmp_path)
+
+    repo.original_cwd = base_repo.original_cwd
+
+    # Register and activate the kubeflow stack
+    orchestrator = KubeflowOrchestrator(
+        name="local_kubeflow_orchestrator",
+        custom_docker_base_image_name="zenml-base-image:latest",
+        synchronous=True,
+    )
+    metadata_store = repo.active_stack.metadata_store.copy(
+        update={"name": "local_kubeflow_metadata_store"}
+    )
+    artifact_store = repo.active_stack.artifact_store.copy(
+        update={"name": "local_kubeflow_artifact_store"}
+    )
+    container_registry = BaseContainerRegistry(
+        name="local_registry", uri="localhost:5000"
+    )
+    kubeflow_stack = Stack(
+        name="local_kubeflow_stack",
+        orchestrator=orchestrator,
+        metadata_store=metadata_store,
+        artifact_store=artifact_store,
+        container_registry=container_registry,
+    )
+    repo.register_stack(kubeflow_stack)
+    repo.activate_stack(kubeflow_stack.name)
+
+    # Provision resources for the kubeflow stack
+    kubeflow_stack.provision()
+
+    yield repo
+
+    # Deprovision the resources after all tests in this module are finished
+    kubeflow_stack.deprovision()
+    os.chdir(str(base_repo.root))
+    shutil.rmtree(tmp_path)
+
+
+@pytest.fixture
+def clean_kubeflow_repo(shared_kubeflow_repo, clean_repo):
+    """Creates a clean repo with a provisioned local kubeflow stack."""
+    # Copy the stack configuration from the shared kubeflow repo. At this point
+    # the stack resources are already provisioned by the module-scoped fixture.
+    kubeflow_stack = shared_kubeflow_repo.active_stack
+    clean_repo.register_stack(kubeflow_stack)
+    clean_repo.activate_stack(kubeflow_stack.name)
+
+    # Delete the artifact store of previous tests
+    if os.path.exists(kubeflow_stack.artifact_store.path):
+        shutil.rmtree(kubeflow_stack.artifact_store.path)
+
+    yield clean_repo
+
+
+def example_runner(examples_dir):
+    """Get the executable that runs examples.
+
+    By default returns the path to an executable .sh file in the
+    repository, but can also prefix that with the path to a shell
+    / interpreter when the file is not executable on its own. The
+    latter option is needed for windows compatibility.
+    """
+    return (
+        [os.environ[SHELL_EXECUTABLE]] if SHELL_EXECUTABLE in os.environ else []
+    ) + [str(examples_dir / EXAMPLES_RUN_SCRIPT)]
+
+
+@pytest.mark.parametrize(
+    "example_configuration",
+    [pytest.param(example, id=example.name) for example in examples],
+)
+def test_run_example_on_kfp(
+    example_configuration: ExampleIntegrationTestConfiguration,
+    clean_kubeflow_repo,
```

Comment:
Could it make sense to combine the kubeflow integration tests with the local integration tests to avoid duplicating the test that have to be written for a new example? Then we could just run test_run_example_on_kfp or test_run_example_locally

Reply:
That was the initial goal, but if we only want to run the kubeflow tests e.g. on develop but the local integration test every time, we'll need to figure out a way to separate them. I have some ideas for that, but wanted to leave the existing stuff intact until both your and my PR are ready/merged

Reply:
```bash
pytest tests/integration/test_examples_on_kubeflow.py::test_run_example_on_kfp
```
That one would only run the kubeflow examples. Similarly we could only run local examples

Reply:
Ah okay I misunderstood the request I think, I thought merging them into a single test function (That would have the additional benefit of potentially only installing the requirements once into the separate virtual environment).If you think it doesn't lead to any complicated merge conflict with your branch(es), I can certainly merge them into a single file and use my example configurations to reduce the duplicated code

Reply:
I actually merged them into a single function and used `pytest.mark.parametrize` for now. That way it's easy to enable kubeflow but we can also quickly split it into two functions if necessary